addons/stock/models/stock_production_lot.py

```diff
@@ -188,7 +188,7 @@ def action_lot_open_transfers(self):
             })
         return action
 
-    def _find_delivery_ids_by_lot(self):
+    def _find_delivery_ids_by_lot(self, lot_path=[]):
```

Comment:
In python we shouldn't put a mutable args as default: 
https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html#anti-pattern 

Also instead of using list it is far more efficient to use a set instead for the `not in` (O(n) vs O(1))